packages/rum-core/src/common/patching/history-patch.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { INVOKE, HISTORY_PUSHSTATE } from '../constants'
+var alreadyPatched = false
+
+export function patchHistory(callback) {
+  if (alreadyPatched || !window.history) {
+    return
+  }
+  alreadyPatched = true
+
+  const nativePushState = history.pushState
+  if (typeof nativePushState === 'function') {
+    history.pushState = function(state, title, url) {
```

Comment:
I believe it's also called for redirects which should be ignored and not captured as part of transaction.

Reply:
I won't start a new transaction only redefines the old one which I think should be ok!

Reply:
Have to be careful here since if the redirect is too slow, we will discard it and create a new one. Its an edge case, but might want to think about reusability threshold.